packages/devtools_app/lib/src/screens/memory/shared/heap/heap_analyzer.dart

```diff
@@ -2,23 +2,53 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'package:flutter/cupertino.dart';
+
 import 'model.dart';
 
-/// Sets the field [retainer] for each object in the [heap], that has retaining
-/// path to the root.
-///
-/// The algorithm takes O(number of references in the heap).
+/// Sets the field retainer and retainedSize for each object in the [heap], that
+/// has retaining path to the root.
 void buildSpanningTree(AdaptedHeap heap) {
-  final root = heap.objects[AdaptedHeap.rootIndex];
-  root.retainer = -1;
+  assert(!heap.isSpanningTreeBuilt);
+  _setRetainers(heap);
+  heap.isSpanningTreeBuilt = true;
+  assert(_verifyHeapIntegrity(heap));
+}
+
+List<HeapStatsRecord> heapStats(AdaptedHeap? heap) {
+  final result = <String, HeapStatsRecord>{};
+  if (heap == null) return [];
+  if (!heap.isSpanningTreeBuilt) buildSpanningTree(heap);
+  for (var object in heap.objects) {
+    // We do not show objects that will be garbage collected soon.
+    if (object.retainedSize == null || object.isSentinel) continue;
```

Comment:
how accurate is this? should we be trying to force a GC before we take a snapshot to make sure?

Reply:
It is accurate (if there is no bugs in gc and snapshot). Even if we force GC, it will be different command (take profile) and new GCable objects can show up. 
Forcing GC affects application, that I would avoid if possible, and it is possible in this case.